rocketpy/motors/fluid.py

```diff
@@ -12,12 +16,18 @@ class Fluid:
     ----------
     name : str
         Name of the fluid.
-    density : float
-        Density of the fluid in kg/m³.
+    density : int, float, callable, string, array, Function
+        Density of the fluid in kg/m³. If a int or float is given,
+        it is considered a constant in time. A callable, csv file
+        or an list of points can be given to express the density
+        as a function of time. The parameter is used as a ``Function``
+        source.
```

Comment:
I believe it makes no sense to have a fluid density depending on the time. How can someone predict these function in the first place?

Wouldn't be more appropiate to have this density parameter depending on the temperature and pressure? To create a function like this:

```python
def density(T, P):
    # so some interpolation
    return value
```

This could be created with the assistance of 2D Function objects and .csv files.

We could even include some "common" fluids csv files in the package, so the user could just call the function with the fluid name and the function would load the csv file and return the 2D Function object. 

In case the user still wants to use a constant value, it would still be possible.

Reply:
Thanks for your suggestion, I considered it at first, but I was hesitant about it for the following reasons. The Tanks module currently does not calculate any thermodynamical quantities.

Correct me if I am mistaken, but unless the Temperature and Pressure in your comment depend on time, we cannot compute anything with it.

At this point, if the Temperature and Pressure are know time variable quantities, the density is also known as a time variable quantity.

Furthermore, the Pressure and Temperature are not used anywhere is Tanks simulation, we would be introducing two new quantities that are not necessary.

I can see two ways of getting the Fluid density in a static fire:

- Having sensors that compute two thermodynamical properties (such as temperature and pressure) and inferring the others from thermodynamical models;
- Having sensors for mass flow and fluid level inside the tank, then the density can be computed.

In my view, in both ways it rather easy to get the density as a function of time. If we implement it from pressure and temperature, we limit ourserves to a particular case (but a really common one) of the first option.

Reply:
Can we make Fluid objects motor-agnostic? If fluid's properties depend on time, this time might be associated with a particular motor, so the same fluid couldn't be used in other tanks.

I was thinking about this: 
- You have the volume of the tank, this is always a fixed value.
- By each time, you calculate the mass of fluid multiplying the volume by the density,
- This density would actually depend on pressure and temperature.

I know that different types of liquid motors will be defined based on different types of input (liquid level, pressure, mass, etc.), but that would be something to be carefully worked on.

What really complicates the whole thing is that we have both liquids and gases defined as a Fluid, but those may present quite different behaviors.